### PR TITLE
 `assign_datetime` algorithm

### PR DESCRIPTION
**Pull request by ramiromagno**
Thank you for your Pull Request! We have developed this task checklist from the
[Development Process
Guide](https://pharmaverse.github.io/sdtm.oak/CONTRIBUTING.html)
to help with the final steps of the process. Completing the below tasks helps to
ensure our reviewers can maximize their time on your code as well as making sure
the oak codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task
or check off that it is not relevant to your Pull Request. This checklist is
part of the Github Action workflows and the Pull Request will not be merged into
the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull z
Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the 
[tidyverse style guide](https://style.tidyverse.org/). Run 
`styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should
consider realistic data scenarios and edge cases, e.g. empty datasets, errors,
boundary cases etc. - See 
[Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you 
fully follow the 
[deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords
and families. Refer to the 
[categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the
`NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it
has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build oak site `pkgdown::build_site()` and check that all affected
examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/assign_datetime.R [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        .format = raw_fmt,
+        .na = raw_unk,
+        .warn = .warn
+      )
+
+    der_dat <-
+      raw_dat |>
+      dplyr::select(c(id_vars, raw_var)) |>
+      dplyr::mutate("{tgt_var}" := tgt_val) |> # nolint object_name_linter()
+      dplyr::select(-rlang::sym(raw_var))
```
Comment (rammprasad) [inline R/assign_datetime.R:155]: This works well with just the date as in the example. We need to update it to make sure it works with the date and time when the input is in two different variables.

```R
cm1 <-
   assign_datetime(
     raw_dat = md1,
     raw_var = c("MDEDR","MDETM"),
     raw_fmt = c("d-m-y", "H:M:S"),
     raw_unk = c("UN", "UNK"),
     tgt_var = "CMSTDTC"
   )
```
Results in error.

This is similar to `create_iso8601(md1$MDEDR, md1$MDETM, .format = c("d-m-y", "H:M:S"), .check_format = FALSE)`

Rest everything looks good to me. Please add a test case as well.

**Review by rammprasad — COMMENTED** [review]
Added my comment

File: R/assign_datetime.R (hunk 1 above)
Comment (ramiromagno) [reply R/assign_datetime.R:155]: Thanks @rammprasad ! It's a silly mistake, `dplyr::select(-rlang::sym(raw_var))` should have been simply `dplyr::select(-raw_var)`. I will fix, and add a test case.

Comment (ramiromagno) [reply R/assign_datetime.R]: @rammprasad: fixed now.

File: R/assign_datetime.R [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+#'     raw_dat = md1,
+#'     raw_var = "MDBDR",
+#'     raw_fmt = "d-m-y",
+#'     raw_unk = c("UN", "UNK"),
+#'     tgt_var = "CMSTDTC"
+#'   )
+#'
+#' cm1
+#' problems(cm1$CMSTDTC)
+#'
```
Comment (rammprasad) [inline R/assign_datetime.R:68]: Can we add this example with date and time?

```R
cm1 <-
   assign_datetime(
     raw_dat = md1,
     raw_var = c("MDEDR","MDETM"),
     raw_fmt = c("d-m-y", "H:M:S"),
     raw_unk = c("UN", "UNK"),
     tgt_var = "CMSTDTC"
   )
```

**Review by rammprasad — APPROVED** [review]
Looks good to me, ready to go.

File: R/assign_datetime.R (hunk 2 above)
Comment (ramiromagno) [reply R/assign_datetime.R:68]: Sure! Will do.

**Review by edgar-manukyan — APPROVED** [review]
Thanks so much @ramiromagno 💯 ⭐ ⭐ ⭐, very nice job. Awesome tests.

I left a super-minor suggestion in case you want to consider.

File: tests/testthat/test-assign_datetime.R [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      9L,      "MD1",       378L,            "26-Jan-20",   "26-Jan-20",   "07:00:00",
+      10L,     "MD1",       378L,            "28-Jan-20",   "1-Feb-20",    NA,
+      11L,     "MD1",       378L,            "12-Feb-20",   "18-Feb-20",   NA,
+      12L,     "MD1",       379L,            "10-UNK-2020", "20-UNK-2020", NA,
+      13L,     "MD1",       379L,            NA,            NA,            NA,
+      14L,     "MD1",       379L,            NA,            "17-Feb-20",   NA
+    )
+
+  warning_msg <-
+    "There were 12 parsing problems\\. Run `problems\\(\\)` on parsed results for details\\."
```
[the remark this answers is not in the document]
Comment (ramiromagno) [reply tests/testthat/test-assign_datetime.R:22]: Will do @edgar-manukya. Thank you for the feedback!
